kornia-py/src/lib.rs

```diff
@@ -9,67 +9,78 @@ mod pointcloud;
+    io_mod.add_function(wrap_pyfunction!(io::jpeg::decode_image_jpeg, &io_mod)?)?;
+    io_mod.add_function(wrap_pyfunction!(io::jpeg::encode_image_jpeg, &io_mod)?)?;
+    // PNG
+    io_mod.add_function(wrap_pyfunction!(io::png::read_image_png_u8, &io_mod)?)?;
+    io_mod.add_function(wrap_pyfunction!(io::png::read_image_png_u16, &io_mod)?)?;
+    io_mod.add_function(wrap_pyfunction!(io::png::write_image_png_u8, &io_mod)?)?;
+    io_mod.add_function(wrap_pyfunction!(io::png::write_image_png_u16, &io_mod)?)?;
+    io_mod.add_function(wrap_pyfunction!(io::png::decode_image_png_u8, &io_mod)?)?;
+    io_mod.add_function(wrap_pyfunction!(io::png::decode_image_png_u16, &io_mod)?)?;
+    // TIFF
+    io_mod.add_function(wrap_pyfunction!(io::tiff::read_image_tiff_u8, &io_mod)?)?;
+    io_mod.add_function(wrap_pyfunction!(io::tiff::read_image_tiff_u16, &io_mod)?)?;
+    io_mod.add_function(wrap_pyfunction!(io::tiff::read_image_tiff_f32, &io_mod)?)?;
+    io_mod.add_function(wrap_pyfunction!(io::tiff::write_image_tiff_u8, &io_mod)?)?;
+    io_mod.add_function(wrap_pyfunction!(io::tiff::write_image_tiff_u16, &io_mod)?)?;
+    io_mod.add_function(wrap_pyfunction!(io::tiff::write_image_tiff_f32, &io_mod)?)?;
+    // JpegTurbo
+    io_mod.add_class::<io::jpegturbo::PyImageDecoder>()?;
+    io_mod.add_class::<io::jpegturbo::PyImageEncoder>()?;
+    io_mod.add_function(wrap_pyfunction!(io::jpegturbo::read_image_jpegturbo, &io_mod)?)?;
+    io_mod.add_function(wrap_pyfunction!(io::jpegturbo::write_image_jpegturbo, &io_mod)?)?;
+    io_mod.add_function(wrap_pyfunction!(io::jpegturbo::decode_image_jpegturbo, &io_mod)?)?;
+    m.add_submodule(&io_mod)?;
+
+    // --- 3. IMGPROC SUBMODULE (Color, Enhance, Histogram, Resize) ---
+    let imgproc_mod = PyModule::new(m.py(), "imgproc")?;
+    imgproc_mod.add_function(wrap_pyfunction!(color::rgb_from_gray, &imgproc_mod)?)?;
+    imgproc_mod.add_function(wrap_pyfunction!(color::rgb_from_rgba, &imgproc_mod)?)?;
+    imgproc_mod.add_function(wrap_pyfunction!(color::rgb_from_bgra, &imgproc_mod)?)?;
+    imgproc_mod.add_function(wrap_pyfunction!(color::bgr_from_rgb, &imgproc_mod)?)?;
+    imgproc_mod.add_function(wrap_pyfunction!(color::gray_from_rgb, &imgproc_mod)?)?;
+    imgproc_mod.add_function(wrap_pyfunction!(enhance::add_weighted, &imgproc_mod)?)?;
+    imgproc_mod.add_function(wrap_pyfunction!(histogram::compute_histogram, &imgproc_mod)?)?;
+    imgproc_mod.add_function(wrap_pyfunction!(resize::resize, &imgproc_mod)?)?;
+    m.add_submodule(&imgproc_mod)?;
+
+    // --- 4. GEOMETRY SUBMODULE (Warp, ICP) ---
+    let geometry_mod = PyModule::new(m.py(), "geometry")?;
+    geometry_mod.add_function(wrap_pyfunction!(warp::warp_affine, &geometry_mod)?)?;
+    geometry_mod.add_function(wrap_pyfunction!(warp::warp_perspective, &geometry_mod)?)?;
+    geometry_mod.add_function(wrap_pyfunction!(icp::icp_vanilla, &geometry_mod)?)?;
+    geometry_mod.add_class::<icp::PyICPConvergenceCriteria>()?;
```

Comment:
ICO can you put u de a 3d module? Soon well refactor from rust upstream that too